File: 121.best-time-to-buy-and-sell-stock.java
// @leet start
class Solution {
	public int maxProfit(int[] prices) {
		int l = 0;
		int r = 0;
		int best = 0;
		while (r < prices.length) {
			if (prices[r] < prices[l]) {
				l = r;
			}
			best = Math.max(best, prices[r] - prices[l]);
			r++;
		}
		return best;
	}
}
// @leet end

File: 121.best-time-to-buy-and-sell-stock.py
# @leet start
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        l=0
        r=0
        ans = 0
        while r < len(prices):
            if prices[r] < prices[l]:
                l = r
            ans = max(ans, prices[r] - prices[l])
            r += 1
        return ans


# @leet end
